src/Fmt_ast.ml

```diff
@@ -303,18 +303,61 @@ let rec fmt_longident (li: Longident.t) =
       cbox 0 (fmt_longident li1 $ wrap "(" ")" (fmt_longident li2))
 
 
-let fmt_constant const =
+let fmt_constant (c: Conf.t) const =
+  let is_printable char_code = 0x20 <= char_code && char_code <= 0x7E in
+  let is_not_term_friendly char_code =
```

Comment:
Then it's not minimal any more :)

By me, we don't need to escape any of those since text editors (and tools/compilers) can handle them properly. 

The special case of `\x00` is indeed maybe better escaped because it breaks some Unix tools (`sed` & co are in undefined behavior when there is a NUL).

AFAIK, we're lucky control-codes cannot be part of an UTF-8 character :)

Reply:
I guess that I'm thinking of the mode where ocamlformat produces output to stdout, and I'm not at all confident that sending control codes to stdout will work as expected everywhere. But I'm not an expert in such topics. I was just thinking that in standard (7 bit) ascii there are only control codes and printing characters, and it would be reasonable to escape everything except the printing characters. But I don't have a strong position if you think that sending control codes to stdout will work.

Reply:
:+1:  let's play safe :)